### ISSUE TITLE
support for mtgjson v4 (we still use v2)

### ISSUE THREAD

**Issue by tooomm** [Cockatrice/Cockatrice#3406]
Advantages as far as I can tell (please correct me and expand the list):
- unique ids
- information on legality
- foreign language support
- token
- spoiler

**Comment by Daenyth** [comment]
I think step 1 will be updating oracle to be able to handle either kind of file

**Comment by ctrlaltca** [comment]
About unique ids: i did it some time ago, see #2545.
The problem is that cockatrice is merging cards having the same name as a single card, so you can't really have "unique ids".

**Comment by ShrewMaster** [comment]
Curious Cockatrice user here: how is this going? I wanted to add new cards from the Game Night set, but this set is not (yet?) available on MTGJSON v3. It is available on the new MTGJSON v4, though. Do you have a prediction about when Oracle will be updated to use the new MTGJSON website?

**Comment by ZeldaZach** [comment]
This ticket should be advanced to super high priority. V3 cannot be built any longer so we should look at v4 compatibility ASAP.

Tagging @ctrlaltca @ebbit1q

**Comment by Daenyth** [comment]
I'm probably going to be busy this weekend, but I should be able to make time for this in 2 weeks

**Comment by ShrewMaster** [comment]
Thank you very much, guys! I've been keeping up with your thread on developing the new oracle to support mtgjson 4, and it seems you finished working on the issue. When can we expect this to be released? Or should we try one of your recent developmental versions of Cockatrice?

**Comment by tooomm** [comment]
@ShrewMaster The release got published in the meantime.
Check out the updater in the client or download from our page.
We are still working on reducing parsing time.

<br>

I tried to compile an overview after #3458 added general support
for basic card and set data and what needs to get addressed next:
- uuid
- extract rarity (to use for deck editor filter)
- foreign names (for card art downloads for example; maybe also foreign texts)
- [tokens](https://github.com/Cockatrice/Cockatrice/pull/3458#issuecomment-443458813)
- spoiler
- [scryfall uuid](https://github.com/Cockatrice/Cockatrice/pull/3458#issuecomment-450527571)

**Comment by ctrlaltca** [comment]
We currently support v4 and any property in it.
The remaining  points from @tooomm:
- uuid: supported, but superseded by scryfall uuid
- mtgjson version: not in v4 (in external file), see #3647
- legality (to filter for standard, modern etc.): supported
- foreign names: foreign card art is supported; for foreign text see #2122
- [!back!]: supported
- [tokens]: supported, but we deliberately prefer our source right now
- spoiler: no spoilers in mtgjson afaik
- [scryfall uuid]: supported

**Comment by tooomm** [comment]
>spoiler: no spoilers in mtgjson afaik

There are! At least that's how I understood it @ZeldaZach 

There is also a potential problem since spoiler.xml and cards.xml could contain the same data. But our own spoiler data is updated daily, while mtgjson updates very irregularly.
